### DISCUSSION TITLE
Extending / Cloning Layouts

### DISCUSSION THREAD

**Discussion by nodeworks** [directus/directus#12428] [Feature Requests]
Wondering if it's possible (now or in the future) to extend a layout such as Tabular. The problem with cloning into a custom extension is that you don't have access to the "app" repo that the imports reference:

```@/utils/notify```

This is because the @directus/app repo only has the bundled and built code and not the source files so we can't actually import these files.

**Comment by rijkvanzanten** [comment]
I think the best approach would be to add a bunch of slots in the layouts themselves, so you could re-use the existing layout in a custom layout, and extend it based on known slots. For example:

```vue
<layout-tabular>
  <template #prepend>
    <v-button>Click me!</v-button>
  </template>
</layout-tabular>
```

🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
